core/src/main/scala/com/pingcap/tispark/utils/WriteUtil.scala

```diff
@@ -225,37 +225,30 @@ object WriteUtil {
       index: TiIndexInfo,
       tiTableInfo: TiTableInfo,
       remove: Boolean): (SerializableKey, Array[Byte]) = {
-
     // NULL is only allowed in unique key, primary key does not allow NULL value
-    val encodeResult = IndexKey.encodeIndexDataValues(
-      row,
-      index.getIndexColumns,
-      handle,
-      index.isUnique && !index.isPrimary,
-      tiTableInfo)
-    val indexKey = IndexKey.toIndexKey(
-      locatePhysicalTable(row, tiTableInfo),
-      index.getId,
-      encodeResult.keys: _*)
-
+    val encodeResult =
+      IndexKey.genIndexKey(locatePhysicalTable(row, tiTableInfo), row, index, handle, tiTableInfo)
     val value = if (remove) {
       new Array[Byte](0)
     } else {
-      if (encodeResult.appendHandle) {
-        val value = new Array[Byte](1)
-        value(0) = '0'
-        value
+      if (!handle.isInt) {
```

Comment:
can you add some explanation for your changed logic.
for example: when appendHandle is true and handle.isInt is false, the logic is diffierent.

Reply:
I add some explain here.